006-empty-interface-and-type-assertion/empty_interface.go

```diff
@@ -6,21 +6,50 @@ package emptyint
 // If the input list contains any other type, that element of the list is ignored.
 // If the list is empty (or consists of only non-supported types) returns 0.0
 func MultitypeSum(input []interface{}) float64 {
-	return 0.0 // TODO implement
+	multiSum := float64(0)
+	for _, value := range input {
+		switch val := value.(type) {
+		case float64:
+			multiSum += val
+		case int64:
+			multiSum += float64(val)
+		case uint64:
+			multiSum += float64(val)
+		default:
+		}
+	}
+	return multiSum
 }
 
 // Stringer is an interface for something which can be represented as a string
 // Do not modify this interface!
 type Stringer interface {
-
 	// String gets the string representation of this Stringer
 	String() string
 }
 
+type defaultStringer struct {
+	str string
+}
+
+func (ds defaultStringer) String() string {
+	return ds.str
+}
+func (ds defaultStringer) AddString(str string) {
+	ds.str += str
+}
+
 // AppendIfStringer checks if "input" is an implementation of Stringer and, if yes, returns a new Stringer
 // with the value of "str" appended to the output of input.String():
 //		inputIfStringer.String() + str
 // If "input" is not a Stringer, returns a new Stringer containing only the value of "str".
 func AppendIfStringer(input interface{}, str string) Stringer {
-	return nil // TODO implement
+	if input == nil {
```

Comment:
The type assertion will also catch the `nil` case, so no need for the `if` statement.